### ISSUE TITLE
Label disappear if rounded with round(..., 0)

### ISSUE THREAD

**Issue by apicella2** [qgis/QGIS#36467]
QGIS 3.12.0, Windows 10
Shapefile joined with Excel Sheet
Rule-based Labeling
Label Text:
- round("joined Field recognized as String", **-1** )... works fine
- round("joined Field recognized as String", **1** )... works fine
- round("joined Field recognized as String", **0** )... Label not shown

**Comment by gioman** [comment]
@apicella2 please attach a small sample project+data.

**Comment by nyalldawson** [comment]
@gioman why was stale/feedback removed here? We never got a reproducible test case...

**Comment by apicella2** [comment]
Hi @gioman, hi @nyalldawson 
I'm sorry, I disappeared like the labels...

I've just created a small test with the last version of QGIS (3.18.3):
[test_qgis_labels.zip](https://github.com/qgis/QGIS/files/6520573/test_qgis_labels.zip)

There is a line, joined with a table.
The table has a field containing some numbers, but the field is a string.

I will show that field, by rounding the number.
This means, the string must be recognized as real number (when present).
If I round it with a decimal place, it works well. The text is recognized as number.
If I round it as integer, the label is not showed.

If it works in a case, it should work in other one as well. Otherwise, it shouldn't work at all.

Thank you very much, bye.
